lms/djangoapps/course_api/views.py

```diff
@@ -2,52 +2,122 @@
 Course API Views
 """
 
-from rest_framework.views import APIView
-from rest_framework.response import Response
-from rest_framework.reverse import reverse
+from rest_framework.exceptions import NotFound, PermissionDenied
+from rest_framework.views import APIView, Response
 
+from opaque_keys import InvalidKeyError
 from opaque_keys.edx.keys import CourseKey
+
+
 from openedx.core.lib.api.view_utils import view_auth_classes
-from xmodule.modulestore.django import modulestore
+
+from .api import course_detail, list_courses
 
 
 @view_auth_classes()
-class CourseView(APIView):
+class CourseDetailView(APIView):
     """
-    Course API view
+    View class to describe a course
     """
 
     def get(self, request, course_key_string):
         """
         Request information on a course specified by `course_key_string`.
-            Body consists of a `blocks_url` that can be used to fetch the
-            blocks for the requested course.
+
+        Body consists of the following fields:
+            * `blocks_url`: can be used to fetch the blocks
+            * `course_image`:
+            * `description`:
+            * `end`: Date the course ends
+            * `enrollment_end`: Date enrollment ends
+            * `enrollment_start`: Date enrollment begins
+            * `id`: Course key
+            * `name`: Name of the course
+            * `number`: Catalog number of the course
+            * `org`: Name of the organization that owns the course
+            * `start`: Date the course begins
+            * `start_display`: Readably formatted start of the course
+            * `start_type`: Hint describing how `start_display` is set. One of:
+                  - `"string"`: manually set
+                  - `"timestamp"`: generated form `start` timestamp
+                  - `"empty"`: the start date should not be shown
 
         Arguments:
-            request (HttpRequest)
-            course_key_string
+            request: (HttpRequest)
+            course_key_string: (string) Key for the desired course
+
+        Parameters:
+            username (optional):
+                The username of the specified user whose visible courses we
+                want to see.  Defaults to the current user.
```

Comment:
Whether the course is visible to the user or not.  Specifically, if the user doesn't have `see_exists` permission on the course, they should not see results for it.  This might be the case if a course is by invitation only (and the user hasn't been invited), or if enrollment has not yet begun.